### ISSUE TITLE
No error for no-unneeded-ternary for object 

### ISSUE THREAD

**Issue by aboyton** [eslint/eslint#10492]
**Tell us about your environment**

* **ESLint Version:** 4.10
* **Node Version:** 8.11.1
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?** `babel-eslint`

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const foo = {};
const bar = foo.bar ? foo.bar : 0;
```

**What did you expect to happen?**
I expected to get an error with `no-unneeded-ternary` like the code below does.

```js
const foo = {};
const bar = foo ? foo : 0;
```

**What actually happened? Please include the actual, raw output from ESLint.**
No error.

I'm guessing that this is because the code is only checking for `"Identifier"` and not also `"MemberExpression"`.

```js
function matchesDefaultAssignment(node) {
  return node.test.type === "Identifier" &&
         node.consequent.type === "Identifier" &&
         node.test.name === node.consequent.name;
}
```

**Comment by platinumazure** [comment]
Hi @aboyton, thanks for the issue.

I think we only check for Identifier because a MemberExpression could invoke a getter property with side effects, similar to a function call.

Hoping another team member can provide more context.

**Comment by ilyavolodin** [comment]
I don't think that's the case. I think it was just something that was never implemented.

**Comment by aboyton** [comment]
I'd wondered if it was due to getters and setters. What would you want to implement this? A PR or does it need some votes and a champion first?

In terms of making a PR, are there any helper functions that you have that compare if two member expressions are the same?

**Comment by nzakas** [comment]
@aboyton sorry, I think we lost track of this. To answer your questions: we need a team member to champion this as an enhancement and get some votes ([full process](https://eslint.org/docs/developer-guide/contributing/rule-changes#accepting-a-rule-change)). I'm willing to :+1: but don't have the additional time to champion.

I don't believe that we have a helper function for comparing member expressions, currently.
